internal/dataplane/kong_client.go

```diff
@@ -489,7 +492,14 @@ func (c *KongClient) Update(ctx context.Context) error {
 
 	// In case of a failure in syncing configuration with Gateways, propagate the error.
 	if gatewaysSyncErr != nil {
-		if recoveringErr := c.tryRecoveringFromGatewaysSyncError(ctx, cacheSnapshot, gatewaysSyncErr); recoveringErr != nil {
+		if recoveringErr := c.tryRecoveringFromGatewaysSyncError(
+			ctx,
+			cacheSnapshot,
+			gatewaysSyncErr,
+			// there are actually multiple returned hashes, and there can apparently be _no_ returned hashes,
+			// so this either can't align to those like the other endpoints or needs to pipe it in from elsewhere
+			"TODO",
```

Comment:
I'd suggest dropping the `hash` whatsoever as that is not very relevant here as the fallback operates on the Kubernetes cache stores, not the config generated. We will always display the latest state, I don't think `hash` would be useful for anything from a user's perspective.

Reply:
My thinking is that at present we have a somewhat organically expanding set of debug endpoints, and we need _something_ for users to correlate them. We should consider some more user-friendly sort of ID, but at present the hash is at least readily available as a basic "these things are inter-related" key.

It's very imperfect in that it doesn't really represent the state, where the relationship between the fallback/active/attempted configs isn't really expressed by the hash, but I'd like to have _something_ tying them together beyond "this is the value the endpoint returns at the time you accessed it". Some more holistic evaluation of debug workflows would be ideal to inform how we design the diagnostic endpoints, but as-is it's unfortunately more organic "here's the metadata we have on hand, figure out how to make use of it after" at present. Diagnostic features sadly get the short end of the feature design stick :|

Reply:
Yeah, I see your point. I think that in the current situation (pressure of time before the release + no obvious correlation id) IMO it's better to not use _anything_ we have at hand as that might be incorrect and hard to revert from later once it's shipped.

Reply:
+1 on not shipping anything rather than then trying to amend/revert it afterwards